docs/api.rst

```diff
@@ -171,20 +171,45 @@ Replace each event value with x.::
 
   constant('tick', periodic(1000))
 
+.. _tap:
+
 tap
 ^^^
 
 .. code-block:: haskell
 
   tap :: (a -> *) -> Stream a -> Stream a
 
+Perform a side-effect for each event in stream.
+
+.. code-block:: javascript
+
+  stream:         -a-b-c-d->
+  tap(f, stream): -a-b-c-d->
+
+For each event in stream, f is called, but the value of its result is ignored. 
+If f fails (ie throws), then the returned stream will also fail. The stream 
+returned by tap will contain the same events as the original stream.
+
+.. _ap:
+
 ap
-^^
+^^^
 
 .. code-block:: haskell
 
   ap :: Stream (a -> b) -> Stream a -> Stream b
 
+Apply the latest function in a stream of functions to the latest value in stream.
```

Comment:
Does it matter if it is a "stream of values"? I mean, couldn’t it be a stream of anything? How about:

"Apply the latest emitted function in a stream of functions to the latest emitted value of another stream."